File: solutions/ocaml/050-climbing-stairs.ml
(*
  Problem 50: Climbing Stairs (LeetCode 70)
  Difficulty: Easy
  Language: OCaml
*)
let climb_stairs n =
  let rec go i a b = if i >= n then a else go (i+1) b (a+b)
  in go 0 1 1

File: solutions/rust/050-climbing-stairs.rs
/**
 * Problem 50: Climbing Stairs (LeetCode 70)
 * Difficulty: Easy
 * Language: Rust
 */
pub fn climb_stairs(n: i32) -> i32 {
    (1..n).fold((1, 1), |(a, b), _| (b, a + b)).0
}
